validation/userRegisterValidator.ts

```diff
@@ -0,0 +1,14 @@
+import * as registrationValidator from "yup";
+
+export const registerSchema = registrationValidator.object({
+    userName: registrationValidator.string().trim().required(),
+    email: registrationValidator.string().email().trim().required(),
+    password: registrationValidator
+        .string()
+        .trim()
+        .min(4)
+        .max(10)
+        // Minimum eight and maximum 10 characters, at least one uppercase letter, one lowercase letter, one number and one special characte
+        // .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,10}$/)
```

Comment:
I posted it on Discord and it seems most would be happy with the pattern below to begin with. By adding this pattern you no longer need to add .min(4).max(10) as the pattern will also check length.

// Minimum six and maximum 15 characters, at least one uppercase letter, one lowercase letter, one number and one special character
// .matches(/^(?=.[a-z])(?=.[A-Z])(?=.\d)(?=.[-@$!%?&])[A-Za-z\d-@$!%?&]{6,15}$/)

Reply:
I changed the pattern and "password" to "Abc-1234" in integrationTests.test.tc as well. 
You can use createRandomUserForRegister() to create random user info.